mmdet3d/core/bbox/coders/km3d_decoder.py

```diff
@@ -0,0 +1,413 @@
+        (l*cosµ)/2+(w*sinµ)/2-kp_1x*[-(l*sinµ)/2+(w*cosµ)/2] as the first row.
+        By construction, data_B is the constant involving (l*cosµ)/2+(w*sinµ)/2
+        and its kind; data_C is the constant involving -(l*sinµ)/2+(w*cosµ)/2
+        and its kind.
+        Therefore, matrix_B-kp_norm*matrix_C can result in the final matrix.
+
+    """
+    data_B = torch.FloatTensor([
+        length * 0.5 * cos_rot_y + width * 0.5 * sin_rot_y, height * 0.5,
+        length * 0.5 * cos_rot_y - width * 0.5 * sin_rot_y, height * 0.5,
+        -length * 0.5 * cos_rot_y - width * 0.5 * sin_rot_y, height * 0.5,
+        -length * 0.5 * cos_rot_y + width * 0.5 * sin_rot_y, height * 0.5,
+        length * 0.5 * cos_rot_y + width * 0.5 * sin_rot_y, -height * 0.5,
+        length * 0.5 * cos_rot_y - width * 0.5 * sin_rot_y, -height * 0.5,
+        -length * 0.5 * cos_rot_y - width * 0.5 * sin_rot_y, -height * 0.5,
+        -length * 0.5 * cos_rot_y + width * 0.5 * sin_rot_y, -height * 0.5
+    ])
+
+    matrix_B[:, :, :] = data_B
+
+    data_C = torch.FloatTensor([
+        -length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y
+    ])
+
+    matrix_C[:, :, :] = data_C
+
+    # final matrix_B in shape (1 boxes 16)
+    matrix_right = matrix_B - kp_norm * matrix_C
+
+    # make matrix_left_transposed in shape(batch * boxes 3 16)
+    matrix_left_transposed = matrix_left.permute(0, 1, 3, 2)
+    matrix_left_transposed = matrix_left_transposed.view(batch * boxes, 3, 16)
+
+    matrix_left = matrix_left.view(batch * boxes, 16, 3)
+    matrix_right = matrix_right.view(batch * boxes, 16, 1).float()
+
+    # pinv here in shape (batch * boxes 3 3)
+    pinv = torch.bmm(matrix_left_transposed, matrix_left)
+    pinv = torch.inverse(pinv)
+
+    # pinv here in shape (batch * boxes 3 16)
+    pinv = torch.bmm(pinv, matrix_left_transposed)
+
+    # pinv here in shape (batch * boxes 3 1)
+    pinv = torch.bmm(pinv, matrix_right)
+
+    # final pinv in shape (batch, boxes, 3)
+    pinv = pinv.view(batch, boxes, 3, 1).squeeze(3)
```

Comment:
pinv -> points_3d. Please also avoid abusing the intermediate variables with the same name.